File: signature/jquery_ui.go
package signature

var JQueryUiSignature = Signature{
	Name:        "jquery-ui",
	Description: "jQuery UI is a curated set of user interface interactions, effects, widgets, and themes built on top of the jQuery JavaScript Library.",
	Cpe:         "cpe:/a:jqueryui:jquery_ui",

	DetectPattern: DetectPattern{
		URLs: []string{
			"jquery-ui",
		},
		Bodies: []string{
			"jQuery UI CSS Framework",
			"/*! jQuery UI",
		},
	},
	VersionPattern: VersionPattern{
		URLs: []string{
			`jquery-ui-(\d+\.\d+\.\d+)`,
		},
		Bodies: []string{
			`jQuery UI CSS Framework (\d+\.\d+\.\d+)`,
			`/*! jQuery UI - v(\d+\.\d+\.\d+)`,
		},
	},
}

File: src/signatures/technologies/jquery_ui.ts
import type { Signature } from "../_types.js";
import { jquerySignature } from "./jquery.js";

export const jqueryUiSignature: Signature = {
  name: "jQuery UI",
  description:
    "jQuery UI is a collection of GUI widgets, animated visual effects, and themes implemented with jQuery, Cascading Style Sheets, and HTML.",
  cpe: "cpe:/a:jquery:jquery_ui",
  rule: {
    confidence: "high",
    urls: [
      "jquery-ui[./-]?(\\d+\\.\\d+\\.\\d+)?",
      "(\\d+\\.\\d+\\.\\d+)?/jquery-ui",
      "jquery-ui.*\\.js",
    ],
    javascriptVariables: {
      "jQuery.ui.version": "(\\d+\\.\\d+\\.\\d+)?",
    },
  },
  impliedSoftwares: [jquerySignature.name],
};
